modules/database/funcs.go

```diff
@@ -122,15 +143,25 @@ func (d *Database) CreateOrPatchDBWithLabel(
 	labels map[string]string,
 ) (ctrl.Result, error) {
 
-	db := &mariadbv1.MariaDBDatabase{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      h.GetBeforeObject().GetName(),
-			Namespace: h.GetBeforeObject().GetNamespace(),
-		},
-		Spec: mariadbv1.MariaDBDatabaseSpec{
-			// the DB name must not change, therefore specify it outside the mutuate function
-			Name: d.databaseName,
-		},
+	if d.name == "" {
+		d.name = h.GetBeforeObject().GetName()
```

Comment:
Just a note that this is actually a bug to user the before object name here. https://github.com/openstack-k8s-operators/lib-common/pull/66